src/llv8.cc

```diff
@@ -1311,6 +1311,86 @@ void JSObject::Keys(std::vector<std::string>& keys, Error& err) {
   return;
 }
 
+
+void JSObject::Entries(std::vector<std::pair<Value, Value>>& entries,
+                       Error& err) {
+  entries.clear();
+
+  HeapObject map_obj = GetMap(err);
+
+  Map map(map_obj);
+
+  bool is_dict = map.IsDictionary(err);
+  if (err.Fail()) return;
+
+  if (is_dict) {
+    DictionaryEntries(entries, err);
+  } else {
+    DescriptorEntries(entries, map, err);
+  }
+
+  return;
+}
+
+void JSObject::DictionaryEntries(std::vector<std::pair<Value, Value>>& entries,
```

Comment:
It should be fine and a bit easier to understand. Are you sure there were any performance issues because of copying?